text/0000-cargo-install.md

```diff
@@ -0,0 +1,271 @@
+- Feature Name: N/A
+- Start Date: 2015-07-10
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+Add a new subcommand to Cargo, `install`, which will install `[[bin]]`-based
+packages onto the local system in a Cargo-specific directory.
+
+# Motivation
+
+There has [almost always been a desire][cargo-37] to be able to install Cargo
+packages locally, but it's been somewhat unclear over time what the precise
+meaning of this is. Now that we have crates.io and lots of experience with
+Cargo, however, the niche that `cargo install` would fill is much clearer.
+
+[cargo-37]: https://github.com/rust-lang/cargo/issues/37
+
+Fundamentally, however, Cargo is a ubiquitous tool among the Rust community and
+implementing `cargo install` would facilitate sharing Rust code among its
+developers.  Simple tasks like installing a new cargo subcommand, installing an
+editor plugin, etc, would be just a `cargo install` away. Cargo can manage
+dependencies, versions, updates, etc, itself to make the process as seamless as
+possible.
+
+Put another way, enabling easily sharing code is one of Cargo's fundamental
+design goals, and expanding into binaries is simply an extension of Cargo's core
+functionality.
+
+# Detailed design
+
+The following new subcommand will be added to Cargo:
+
+```
+Install a crate onto the local system
+
+Installing new crates:
+    cargo install [options]
+    cargo install [options] [-p CRATE | --package CRATE] [--vers VERS]
```

Comment:
Why does `cargo install` require a `-p` or `--package` instead of pulling from the registry by default (which I would expect to be the most common use)? Is there an argument parsing issue with  this being `cargo install [options] <crate> [--vers VERS]`?